src/modules/Modal/Modal.js

```diff
@@ -265,6 +279,18 @@ class Modal extends Component {
       <ElementType {...rest} className={classes} style={{ marginTop }} ref={this.handleRef}>
         {Icon.create(closeIconName, { onClick: this.handleClose })}
         {children}
+        {_.isNil(children) && header && ModalHeader.create(header)}
+        {_.isNil(children) && content && ModalContent.create(content)}
+        {_.isNil(children) && actions && ModalActions.create(actions.map((action, i) => {
+          if (action.triggerClose) {
+            const onClick = (callback) => (e) => {
+              if (callback) callback(e, this.props)
+              this.handleClose()
+            }
+            action.onClick = onClick(action.onClick)
+          }
+          return Button.create(action, { key: i })
```

Comment:
Using index for a key is an [antipattern](https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318#.cegqr4uf1).  Instead, we can configure the factory to allow the to pass a key.  See comment on `ModalActions.create` below.